src/Features/Core/Portable/Diagnostics/DiagnosticService.cs

```diff
@@ -210,18 +211,26 @@ private void OnCleared(object sender, EventArgs e)
             RaiseDiagnosticsCleared((IDiagnosticUpdateSource)sender);
         }
 
-        public ImmutableArray<DiagnosticData> GetDiagnostics(
+        public ImmutableArray<DiagnosticData> GetPullDiagnostics(Workspace workspace, ProjectId projectId, DocumentId documentId, object id, bool includeSuppressedDiagnostics, Option2<DiagnosticMode> diagnosticMode, CancellationToken cancellationToken)
+            => GetDiagnostics(workspace, projectId, documentId, id, includeSuppressedDiagnostics, forPullDiagnostics: true, diagnosticMode, cancellationToken);
+
+        public ImmutableArray<DiagnosticData> GetPushDiagnostics(Workspace workspace, ProjectId projectId, DocumentId documentId, object id, bool includeSuppressedDiagnostics, Option2<DiagnosticMode> diagnosticMode, CancellationToken cancellationToken)
+            => GetDiagnostics(workspace, projectId, documentId, id, includeSuppressedDiagnostics, forPullDiagnostics: false, diagnosticMode, cancellationToken);
+
+        private ImmutableArray<DiagnosticData> GetDiagnostics(
             Workspace workspace,
             ProjectId projectId,
             DocumentId documentId,
             object id,
             bool includeSuppressedDiagnostics,
             bool forPullDiagnostics,
+            Option2<DiagnosticMode> diagnosticMode,
             CancellationToken cancellationToken)
         {
             // If this is a pull client, but pull diagnostics is not on, then they get nothing.  Similarly, if this is a
             // push client and pull diagnostics are on, they get nothing.
-            if (forPullDiagnostics != workspace.Options.GetOption(InternalDiagnosticsOptions.LspPullDiagnostics))
+            var isPull = workspace.Options.GetOption(diagnosticMode) == DiagnosticMode.Pull;
+            if (forPullDiagnostics != isPull)
                 return ImmutableArray<DiagnosticData>.Empty;
```

Comment:
if the client wants the diagnostics for a purpose that doesn't match teh current diagnostic mode, then they get nothing.

Reply:
should the comment be updated to match this?